include/slang/syntax/SyntaxNode.h

```diff
@@ -320,4 +356,18 @@ class SLANG_EXPORT SeparatedSyntaxList : public SyntaxListBase {
     span<TokenOrSyntax> elements;
 };
 
+template<typename T>
+SeparatedSyntaxList<T>* deepClone(const SeparatedSyntaxList<T>& node, BumpAllocator& alloc) {
+    SmallVectorSized<TokenOrSyntax, 8> buffer(node.size());
+    for (const auto& ele : node.elems()) {
+        if (ele.isToken()) {
+            buffer.append(ele);
+        }
+        else {
+            buffer.append(slang::syntax::deepClone(ele.node()->template as<T>(), alloc));
```

Comment:
I don't think you need the as<T> here, since you have an overload that will take the SyntaxNode and do the correct thing anyway?